### PR TITLE
Gun/disclose

### PR DESCRIPTION
**Pull request by ftreguer**
Add disclose-explicit-values command [#1499](https://spacelephant.tpondemand.com/entity/1499-cli-or-implementer-la-commande-greater)

### CODE REVIEW

File: src/baseCommand.ts [hunk 1]
```diff
@@ -177,4 +181,25 @@ export abstract class BaseCommand extends Command {
...
+            this.debug(e);
+        } finally {
+            this.actionStop();
+        }
+
+        if (error) {
+            throw error;
+        }
+
+        return callBackResult;
```
Comment (dlecan) [inline src/baseCommand.ts]: Why so complicated code?

This is enough
```typescript
        try {
            return await callback(...args);
        } catch (e) {
            this.debug(e);
            throw e;
        } finally {
            this.actionStop();
        }
```

File: src/api.ts [hunk 2]
```diff
@@ -119,9 +119,9 @@ export class UNSCLIAPI {
             .catch(e => {
```
Comment (dlecan) [inline src/api.ts]: Duplicated error handling, which may be resuse again later.
Please extract an error handler here.

```
.catch(customErrorHandler);
```

File: src/commands/did-resolve.ts [hunk 3]
```diff
@@ -49,7 +50,7 @@ export class DidResolveCommand extends BaseCommand {
         }
 
         if (resolved) {
```
Comment (dlecan) [inline src/commands/did-resolve.ts:52]: Combine `if`

File: src/writeCommand.ts [hunk 4]
```diff
@@ -1,9 +1,37 @@
...
+    protected async applyWalletPredicate(recipientId: string, predicate: (wallet) => boolean) {
+        try {
+            const wallet = await this.api.getWallet(recipientId);
+            return predicate(wallet);
+        } catch (e) {
+            if (e instanceof HttpNotFoundError) {
+                this.info(`Wallet ${recipientId} not found.`);
+                return false;
+            }
+            throw e;
```
Comment (dlecan) [inline src/writeCommand.ts:34]: Again the same `customErrorHandler`?

**Review by dlecan — CHANGES_REQUESTED** [review]

File: src/commands/did-resolve.ts (hunk 3 above)
Comment (ftreguer) [reply src/commands/did-resolve.ts:52]: Not the same error

**Review by dlecan — APPROVED** [review]

**Pull request merged**